meson.build

```diff
@@ -199,6 +199,22 @@ gumjs_extra_deps = []
 gumjs_extra_requires = []
 gumjs_extra_libraries = []
 
+have_gir = false
+
+if cc.has_function('g_thread_set_callbacks', dependencies: [glib_dep])
+  cdata.set('HAVE_FRIDA_GLIB', 1)
+else
+  elf_dep = dependency('libelf')
+  extra_deps += [elf_dep]
+
+  gir = find_program('g-ir-scanner')
```

Comment:
I think this should be keyed off whether we're building Gum as a shared library; nothing's stopping us from building our GLib as a dynamic library as well.